### PR TITLE
Split TimelockAuthorizer into TimelockAuthorizer and TimelockAuthorizerManagement

### LINKED ISSUE #2320: Extract Timelock interface
While IAuthorizer should remain unchanged, we should add `ITimelockAuthorizer` and move the associated functions there.

### PR DESCRIPTION
**Pull request by ylv-io**
# Description

Splits `TimelockAuthorizer` into `TimelockAuthorizer` and `TimelockAuthorizerManagement` to provide more clarity and readability of the code. `TimelockAuthorizerManagement`. Extracts all the external methods of `TimelockAuthorizer` into `ITimelockAuthorizer` interface.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [x] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented, including external interfaces
- [x] Tests are included for all code paths
- [x] The base branch is either `master`, or there's a description of how to merge

## Issue Resolution

Resolves #2320

### CODE REVIEW

File: pkg/vault/contracts/authorizer/TimelockAuthorizer.sol [hunk 1]
```diff
@@ -161,65 +121,15 @@ contract TimelockAuthorizer is IAuthorizer, ReentrancyGuard {
     );
 
     /**
-     * @notice Emitted when a root change is scheduled.
+     * @notice Emitted when a new `delay` is set in order to perform action `actionId`.
      */
-    event RootChangeScheduled(address indexed newRoot, uint256 indexed scheduledExecutionId);
+    event ActionDelaySet(bytes32 indexed actionId, uint256 delay);
```
Comment (nventuro) [inline pkg/vault/contracts/authorizer/TimelockAuthorizer.sol:126]: This events should also be moved to the interface.

**Review by nventuro — APPROVED** [review]
Looks great, thanks a lot @ylv-io! I left some suggestions for small adjustments and the location of some things, but overall everything seems ok.

We may want to to reorder some bits later, but I'd no do that now given this is already a large changeset, and the fact that the order is preserved makes reviewing easier.

File: pkg/vault/contracts/authorizer/TimelockAuthorizerManagement.sol [hunk 2]
```diff
@@ -0,0 +1,635 @@
...
+     * @notice A sentinel value for `where` that will match any address.
+     */
+    address public constant EVERYWHERE = address(-1);
+
+    /**
+     * @notice A constant value for `scheduledExecutionId` that will match any execution Id.
+     * Cancelers assigned to this Id will be able to cancel *any* scheduled action,
+     * which is very useful for e.g. emergency response dedicated teams that analyze these.
+     */
+    uint256 public constant GLOBAL_CANCELER_SCHEDULED_EXECUTION_ID = type(uint256).max;
```
[the remark this answers is not in the document]
Comment (ylv-io) [reply pkg/vault/contracts/authorizer/TimelockAuthorizerManagement.sol:50]: But Solidity doesn't allow constants in interfaces. What I am missing?

**Comment by ylv-io** [comment]
Addressed all the comments. Now it looks even better 😍

File: pkg/vault/contracts/authorizer/TimelockAuthorizer.sol [hunk 3]
```diff
@@ -52,386 +53,103 @@ import "./TimelockExecutionHelper.sol";
  * In fact a number of the TimelockAuthorizer's functions may only be called through a scheduled execution so reentrancy
  * is necessary in order to be able to call these.
  */
-contract TimelockAuthorizer is IAuthorizer, ReentrancyGuard {
-    using Address for address;
+contract TimelockAuthorizer is IAuthorizer, TimelockAuthorizerManagement {
+    // solhint-disable-next-line const-name-snakecase
+    uint256 private constant _max_delay = 2 * (365 days);
```
Comment (nventuro) [inline pkg/vault/contracts/authorizer/TimelockAuthorizer.sol:58]: We typically do `_MAX_DELAY` for things such as these. I don't think we ever use lowercase snakecase.

**Review by nventuro — APPROVED** [review]
Fantastic :grin:

**Pull request merged**